### PR TITLE
psm wrapper layer

### PR DESCRIPTION
**Pull request by Geometer1729**
Closing #13 
uses #12 so targeting that branch for now

### CODE REVIEW

File: src/Hedgehog/Plutus/Model.hs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{-# OPTIONS_GHC -Wno-missing-import-lists #-}
+
+-- I think they are fine here because they are just reexported
```
Comment (AriFordsham) [inline src/Hedgehog/Plutus/Model.hs:3]: Unfortunately style guide says no. You may use wildcard exports, though (these are not allowed for imports)

**Review by AriFordsham — CHANGES_REQUESTED** [review]
Regarding error handling: My opinion is that returning `Either String` is a violation of separation of concerns. Functions with only one reason to fail should return `Maybe` - if this is not obvious it should be commented. If there may be more than one reason for failure, best is `Either` over a custom type. Describing the error should be left to the highest level.

Otherwise excellent so far.

File: src/Hedgehog/Plutus/Model.hs (hunk 1 above)
Comment (Geometer1729) [reply src/Hedgehog/Plutus/Model.hs]: Fair enough :+1:

**Comment by Geometer1729** [comment]
Should I still re-throw those errors in `submitTx`? I assume an error like these should cause the test to fail.

**Pull request closed without merging**